### PR TITLE
feat(daemon): remote flow on star routing

### PR DESCRIPTION
**Pull request by bwanglzu**
- [x] get arguments
- [x] flow get all/clear
- [x] flow c/r/u/d sync/async
- [x] above with replicas & shards *need to fix env setting*
- [x] scalable flow *need to fix env setting* ~deleting a scaled flow is not possible??~
- [x] rolling update
- [x] merge flow env test into `test_remote_flow`
- [ ] port-mapping clean up
- [x] clean up distributed test structure
- [ ] re-enable distributed tests in CI & fix failing tests
- [x] re-enable unit tests in CI & fix failing tests

### CODE REVIEW

File: tests/distributed/test_remote_flows/test_remote_flows.py [hunk 1]
```diff
@@ -31,19 +32,23 @@ def async_jinad_client():
 
 
 def test_remote_jinad_flow(flow_args, jinad_client):
-    payload = replace_enum_to_str(ArgNamespace.flatten_to_dict(flow_args))
+    os.environ['JINA_PROTOCAL'] = 'http'
```
Comment (JoanFM) [inline tests/distributed/test_remote_flows/test_remote_flows.py:35]:
```suggestion
    os.environ['JINA_PROTOCOL'] = 'http'
```

File: jina/peapods/peas/jinad.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ def filepaths(self) -> List[Path]:
             for path in self.args.upload_files:
                 try:
                     fullpath = Path(complete_path(path))
-                    paths.add(fullpath)
+                    paths.add(fullpath.stem)
```
Comment (deepankarm) [inline jina/peapods/peas/jinad.py:176]: Any reason, this is changed?

File: tests/daemon/unit/api/endpoints/partial/test_flow.py [hunk 3]
```diff
@@ -33,6 +28,6 @@ def test_flow_api(monkeypatch, partial_flow_client):
     assert create_response
     assert get_response
     assert get_response.json()['arguments']['port_expose'] == 56789
-    assert endpoint_responses[0].docs[0].content == 'https://jina.ai'
     assert rolling_update_response.status_code == 200
     assert delete_response
+    partial_flow_client.wait_shutdown()
```
Comment (deepankarm) [inline tests/daemon/unit/api/endpoints/partial/test_flow.py:33]: Why is this required?

**Review by deepankarm — CHANGES_REQUESTED** [review]
What all tests are re-enabled in this PR? 
- We need to make sure all tests under `test_topologies`, `test_topologies_docker` are also tested in addition to Flow level tests.
- Components previously needed with 2.x (and no more needed) needs to be removed as part of this PR. 
- Please add a test for [this issue](https://github.com/jina-ai/jina/issues/3941). The code for the test already exists in the issue description.

File: jina/peapods/peas/jinad.py (hunk 2 above)
Comment (bwanglzu) [reply jina/peapods/peas/jinad.py:176]: jinad accpets paths of List[str], here we send `List[Path]`

**Comment by bwanglzu** [comment]
@deepankarm shall I limit the scope of the PR :) when fixing test I found a lot of tests are dependent from each other..in the end to make A work I have to fix B..so enabled all tests.

**Comment by bwanglzu** [comment]
I'll create a fresh PR.
